rust/src/lib.rs

```diff
@@ -1268,25 +1392,27 @@ pub unsafe extern "C" fn zcashlc_derive_transparent_private_key_from_seed(
         let network = parse_network(network_id)?;
         let seed = slice::from_raw_parts(seed, seed_len);
         let account = if account >= 0 {
-            account as u32
+            AccountId::from(account as u32)
         } else {
             return Err(format_err!("account argument must be positive"));
         };
 
-        let index = if index >= 0 {
+        let _index = if index >= 0 {
             index as u32
         } else {
             return Err(format_err!("index argument must be positive"));
         };
-        let sk = derive_secret_key_from_seed(&network, &seed, AccountId(account), index).unwrap();
-        let sk_wif = Wif::from_secret_key(&sk, true);
+        
+        let account_priv_key = extended_private_key_from_seed(&network, seed, account);
+        
+        let encoded = serialize_extended_priv_key(&account_priv_key.unwrap());
 
-        Ok(CString::new(sk_wif.0.to_string()).unwrap().into_raw())
+        Ok(CString::new(hex::encode(encoded)).unwrap().into_raw())
```

Comment:
Since this changes the encoding, is there any risk that the old encoded version might have been persisted anywhere?

Reply:
Fortunately we don't persist it, but there's a good thing to document just in case anyone has.